modules/raft/src/main/java/org/apache/ignite/raft/jraft/rpc/ReadActionRequest.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.raft.jraft.rpc;
+
+import org.apache.ignite.internal.raft.ReadCommand;
+import org.apache.ignite.network.annotations.Transferable;
+import org.apache.ignite.raft.jraft.RaftMessageGroup.RpcActionMessageGroup;
+
+/**
+ * Submit an action to a replication group.
+ */
+@Transferable(RpcActionMessageGroup.READ_ACTION_REQUEST)
+public interface ReadActionRequest extends ActionRequest {
+    /**
+     * @return Action's command.
+     */
+    ReadCommand command();
+
+    /**
+     * @return {@code True} for linearizable reading.
```

Comment:
I don't quite understand this description, could you expand on it? I see that there is a command for reading, can they be not only for reading? How can they **not** be safe reading?

Reply:
I copied the description from the old interface

Reply:
I understand, but you added a new interface, let’s improve it!